src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -2246,6 +2269,34 @@ class Simplex_tree {
   /** \brief Upper bound on the dimension of the simplicial complex.*/
   int dimension_;
   bool dimension_to_be_lowered_ = false;
+
+  // MULTIPERS STUFF
+ public:
+  /**
+   * \brief Sets the number of parameters of the filtrations if SimplexTreeOptions::is_multi_parameter. 
+   * */
+  void set_number_of_parameters(int num) { 
+    static_assert(SimplexTreeOptions::is_multi_parameter, 
+      "Cannot set number of parameters of 1-parameter simplextree."
+    ); 
+    number_of_parameters_ = num; 
+  }
+  /**
+   * \brief Gets the number of parameters of the filtrations if SimplexTreeOptions::is_multi_parameter. 
+   * */
+  int get_number_of_parameters() const { 
+    if constexpr (SimplexTreeOptions::is_multi_parameter)
+      return number_of_parameters_;
+    else
+      return 1;
+  }
+
+  inline static Filtration_value inf_ = std::numeric_limits<Filtration_value>::has_infinity ? 
+      std::numeric_limits<Filtration_value>::infinity() 
+    : std::numeric_limits<Filtration_value>::max(); /**< Default infinite value. */
+
+ private:
+  int number_of_parameters_; /**< Number of parameters of the multi-filtrations when SimplexTreeOptions::is_multi_parameter.-*/
```

Comment:
`number_of_parameters_` default value is 0. TBD, but wouldn't it be clearer from a user point of view to set it to 2 by default ?

Reply:
Yes, why not, as the get number of parameters will always be  1 for non-multiparam simplextrees, this default value seems good.

Reply:
OTOH, a "broken" value like 0 (or even -1) forces the user to be explicit (which can be seen as good or bad).